graph/src/data_source/mod.rs

```diff
@@ -206,6 +206,13 @@ impl<C: Blockchain> DataSource<C> {
             Self::Offchain(_) => vec![],
         }
     }
+
+    pub fn causality_region(&self) -> CausalityRegion {
```

Comment:
If not covered already then adding a simple test would be good

Reply:
I've added a `data_source_helpers` test.